[BUG] Error in Logfile - Traceroutes

In my docker-compose logs, I see:

```
meshmonitor           | [INFO] 🗺️ Traceroutl response from !b29fb37c: {            0                                                                              
 eshmonitor           |   "route": [                                                                                                                              
meshmonitor           |     2733351748                                                                                                                            
meshmonitor           |   ],                                                                                                                                      
meshmonitor           |   "snrTowards": [                                                                                                                         
meshmonitor           |     2,                                                                                                                                    
meshmonitor           |     -58                                                                                                                                   
meshmonitor           |   ],                                                                                                                                      
meshmonitor           |   "routeBack": [                                                                                                                          
meshmonitor           |     2687838236,                                                                                                                           
meshmonitor           |     2732684752                                                                                                                            
meshmonitor           |   ],                                                                                                                                      
meshmonitor           |   "snrBack": [                                                                                                                            
meshmonitor           |     -13,                                                                                                                                  
meshmonitor           |     25,                                                                                                                                   
meshmonitor           |     -43                                                                                                                                   
meshmonitor           |   ]                                                                                                                                       
meshmonitor           | }                                                                                                                                         
meshmonitor-postgres  | 2026-01-20 14:59:34.456 EST [609] ERROR:  invalid input syntax for type integer: "true"
meshmonitor-postgres  | 2026-01-20 14:59:34.456 EST [609] CONTEXT:  unnamed portal parameter $1 = '...'
meshmonitor-postgres  | 2026-01-20 14:59:34.456 EST [609] STATEMENT:                                                                                              
meshmonitor-postgres  |                   UPDATE auto_traceroute_log                                                                                              
meshmonitor-postgres  |                   SET success = $1                                                                                                        
meshmonitor-postgres  |                   WHERE id = (                                                                                                            
meshmonitor-postgres  |                     SELECT id FROM auto_traceroute_log                                                                                    
meshmonitor-postgres  |                     WHERE to_node_num = $2 AND success IS NULL                                                                            
meshmonitor-postgres  |                     ORDER BY timestamp DESC                                                                                               
meshmonitor-postgres  |                     LIMIT 1                                                                                                               
meshmonitor-postgres  |                   )                                                                                                                       
meshmonitor-postgres  |                                                                                                                                           
meshmonitor           | [ERROR] [DatabaseService] Failed to update auto traceroute result async: error: invalid input syntax for type integer: "true"    
```

--- a/src/db/schema/mysql-create.ts
+++ b/src/db/schema/mysql-create.ts
@@ -317,14 +317,20 @@ export const MYSQL_SCHEMA_SQL = `
   ) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_unicode_ci;
 
   CREATE TABLE IF NOT EXISTS upgrade_history (
-    id INT AUTO_INCREMENT PRIMARY KEY,
-    fromVersion VARCHAR(255) NOT NULL,
-    toVersion VARCHAR(255) NOT NULL,
-    upgradeType VARCHAR(255) NOT NULL,
-    status VARCHAR(255) NOT NULL,
-    startedAt BIGINT NOT NULL,
+    id VARCHAR(64) PRIMARY KEY,
+    fromVersion VARCHAR(32) NOT NULL,
+    toVersion VARCHAR(32) NOT NULL,
+    deploymentMethod VARCHAR(32) NOT NULL,
+    status VARCHAR(32) NOT NULL,
+    progress INT DEFAULT 0,
+    currentStep VARCHAR(255),
+    logs TEXT,
+    backupPath VARCHAR(512),
+    startedAt BIGINT,
     completedAt BIGINT,
-    error TEXT
+    initiatedBy VARCHAR(255),
+    errorMessage TEXT,
+    rollbackAvailable BOOLEAN
   ) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_unicode_ci;
 
   CREATE TABLE IF NOT EXISTS custom_themes (

--- a/src/db/schema/postgres-create.ts
+++ b/src/db/schema/postgres-create.ts
@@ -296,14 +296,20 @@ export const POSTGRES_SCHEMA_SQL = `
   );
 
   CREATE TABLE IF NOT EXISTS upgrade_history (
-    id SERIAL PRIMARY KEY,
+    id TEXT PRIMARY KEY,
     "fromVersion" TEXT NOT NULL,
     "toVersion" TEXT NOT NULL,
-    "upgradeType" TEXT NOT NULL,
-    status TEXT NOT NULL DEFAULT 'pending',
-    "startedAt" BIGINT NOT NULL,
+    "deploymentMethod" TEXT NOT NULL,
+    status TEXT NOT NULL,
+    progress INTEGER DEFAULT 0,
+    "currentStep" TEXT,
+    logs TEXT,
+    "backupPath" TEXT,
+    "startedAt" BIGINT,
     "completedAt" BIGINT,
-    error TEXT
+    "initiatedBy" TEXT,
+    "errorMessage" TEXT,
+    "rollbackAvailable" BOOLEAN
   );
 
   CREATE TABLE IF NOT EXISTS custom_themes (

--- a/src/server/migrations/052_fix_upgrade_history_schema.ts
+++ b/src/server/migrations/052_fix_upgrade_history_schema.ts
@@ -1,0 +1,185 @@
+/**
+ * Migration 052: Fix upgrade_history table schema
+ *
+ * The upgrade_history table schema in postgres-create.ts and mysql-create.ts
+ * was out of sync with the Drizzle schema in misc.ts. This migration fixes:
+ *
+ * 1. Renames 'upgradeType' to 'deploymentMethod'
+ * 2. Renames 'error' to 'errorMessage'
+ * 3. Adds missing columns: progress, currentStep, logs, backupPath, initiatedBy, rollbackAvailable
+ * 4. Changes id from SERIAL/INT to TEXT/VARCHAR(64)
+ *
+ * For SQLite: No changes needed as it uses the Drizzle schema directly
+ */
+
+import type Database from 'better-sqlite3';
+import { logger } from '../../utils/logger.js';
+
+export const migration = {
+  up: (_db: Database.Database): void => {
+    logger.debug('Running migration 052: Fix upgrade_history schema...');
+    // SQLite schema is managed by Drizzle and already correct
+    logger.debug('✅ Migration 052: SQLite schema is already correct');
+    logger.debug('✅ Migration 052 completed successfully');
+  },
+
+  down: (_db: Database.Database): void => {
+    logger.debug('Reverting migration 052: No-op for SQLite');
+  }
+};
+
+/**
+ * PostgreSQL migration: Fix upgrade_history table schema
+ */
+export async function runMigration052Postgres(client: import('pg').PoolClient): Promise<void> {
+  logger.debug('Running migration 052 (PostgreSQL): Fix upgrade_history schema');
+
+  try {
+    // Check if table exists
+    const tableExists = await client.query(`
+      SELECT EXISTS (
+        SELECT FROM information_schema.tables
+        WHERE table_schema = 'public' AND table_name = 'upgrade_history'
+      )
+    `);
+
+    if (!tableExists.rows[0].exists) {
+      logger.debug('upgrade_history table does not exist, skipping migration');
+      return;
+    }
+
+    // Check if we need to migrate (does upgradeType column exist?)
+    const needsMigration = await client.query(`
+      SELECT EXISTS (
+        SELECT FROM information_schema.columns
+        WHERE table_schema = 'public'
+          AND table_name = 'upgrade_history'
+          AND column_name = 'upgradeType'
+      )
+    `);
+
+    if (!needsMigration.rows[0].exists) {
+      // Check if deploymentMethod already exists (schema is already correct)
+      const hasDeploymentMethod = await client.query(`
+        SELECT EXISTS (
+          SELECT FROM information_schema.columns
+          WHERE table_schema = 'public'
+            AND table_name = 'upgrade_history'
+            AND column_name = 'deploymentMethod'
+        )
+      `);
+
+      if (hasDeploymentMethod.rows[0].exists) {
+        logger.debug('upgrade_history schema is already correct, skipping');
+        return;
+      }
+    }
+
+    // Drop existing data and recreate table with correct schema
+    // The upgrade_history table typically has temporary data that can be cleared
+    logger.debug('Recreating upgrade_history table with correct schema...');
+
+    await client.query('DROP TABLE IF EXISTS upgrade_history CASCADE');
+
+    await client.query(`
+      CREATE TABLE upgrade_history (
+        id TEXT PRIMARY KEY,
+        "fromVersion" TEXT NOT NULL,
+        "toVersion" TEXT NOT NULL,
+        "deploymentMethod" TEXT NOT NULL,
+        status TEXT NOT NULL,
+        progress INTEGER DEFAULT 0,
+        "currentStep" TEXT,
+        logs TEXT,
+        "backupPath" TEXT,
+        "startedAt" BIGINT,
+        "completedAt" BIGINT,
+        "initiatedBy" TEXT,
+        "errorMessage" TEXT,
+        "rollbackAvailable" BOOLEAN
+      )
+    `);
+
+    logger.debug('✅ Migration 052 (PostgreSQL): upgrade_history table recreated with correct schema');
+  } catch (error) {
+    logger.error('Migration 052 (PostgreSQL) failed:', error);
+    throw error;
+  }
+}
+
+/**
+ * MySQL migration: Fix upgrade_history table schema
+ */
+export async function runMigration052Mysql(pool: import('mysql2/promise').Pool): Promise<void> {
+  logger.debug('Running migration 052 (MySQL): Fix upgrade_history schema');
+
+  try {
+    const connection = await pool.getConnection();
+    try {
+      // Check if table exists
+      const [tables] = await connection.query(`
+        SELECT TABLE_NAME FROM information_schema.tables
+        WHERE TABLE_SCHEMA = DATABASE() AND TABLE_NAME = 'upgrade_history'
+      `);
+
+      if ((tables as any[]).length === 0) {
+        logger.debug('upgrade_history table does not exist, skipping migration');
+        return;
+      }
+
+      // Check if we need to migrate (does upgradeType column exist?)
+      const [columns] = await connection.query(`
+        SELECT COLUMN_NAME FROM information_schema.columns
+        WHERE TABLE_SCHEMA = DATABASE()
+          AND TABLE_NAME = 'upgrade_history'
+          AND COLUMN_NAME = 'upgradeType'
+      `);
+
+      if ((columns as any[]).length === 0) {
+        // Check if deploymentMethod already exists
+        const [deploymentCol] = await connection.query(`
+          SELECT COLUMN_NAME FROM information_schema.columns
+          WHERE TABLE_SCHEMA = DATABASE()
+            AND TABLE_NAME = 'upgrade_history'
+            AND COLUMN_NAME = 'deploymentMethod'
+        `);
+
+        if ((deploymentCol as any[]).length > 0) {
+          logger.debug('upgrade_history schema is already correct, skipping');
+          return;
+        }
+      }
+
+      // Drop and recreate table with correct schema
+      logger.debug('Recreating upgrade_history table with correct schema...');
+
+      await connection.query('DROP TABLE IF EXISTS upgrade_history');
+
+      await connection.query(`
+        CREATE TABLE upgrade_history (
+          id VARCHAR(64) PRIMARY KEY,
+          fromVersion VARCHAR(32) NOT NULL,
+          toVersion VARCHAR(32) NOT NULL,
+          deploymentMethod VARCHAR(32) NOT NULL,
+          status VARCHAR(32) NOT NULL,
+          progress INT DEFAULT 0,
+          currentStep VARCHAR(255),
+          logs TEXT,
+          backupPath VARCHAR(512),
+          startedAt BIGINT,
+          completedAt BIGINT,
+          initiatedBy VARCHAR(255),
+          errorMessage TEXT,
+          rollbackAvailable BOOLEAN
+        ) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_unicode_ci
+      `);
+
+      logger.debug('✅ Migration 052 (MySQL): upgrade_history table recreated with correct schema');
+    } finally {
+      connection.release();
+    }
+  } catch (error) {
+    logger.error('Migration 052 (MySQL) failed:', error);
+    throw error;
+  }
+}

--- a/src/services/database.ts
+++ b/src/services/database.ts
@@ -55,6 +55,7 @@ import { migration as autoTracerouteColumnMigration } from '../server/migrations
 import { migration as notificationChannelSettingsMigration, runMigration049Postgres, runMigration049Mysql } from '../server/migrations/049_add_notification_channel_settings.js';
 import { migration as channelDatabaseMigration, runMigration050Postgres, runMigration050Mysql } from '../server/migrations/050_add_channel_database.js';
 import { migration as decryptedByMessagesMigration, runMigration051Postgres, runMigration051Mysql } from '../server/migrations/051_add_decrypted_by_to_messages.js';
+import { migration as upgradeHistorySchemaMigration, runMigration052Postgres, runMigration052Mysql } from '../server/migrations/052_fix_upgrade_history_schema.js';
 import { validateThemeDefinition as validateTheme } from '../utils/themeValidation.js';
 
 // Drizzle ORM imports for dual-database support
@@ -880,6 +881,7 @@ class DatabaseService {
     this.runNotificationChannelSettingsMigration();
     this.runChannelDatabaseMigration();
     this.runDecryptedByMessagesMigration();
+    this.runUpgradeHistorySchemaMigration();
     this.ensureAutomationDefaults();
     this.warmupCaches();
     this.isInitialized = true;
@@ -1888,6 +1890,26 @@ class DatabaseService {
       logger.debug('✅ Decrypted_by messages migration completed successfully');
     } catch (error) {
       logger.error('❌ Failed to run decrypted_by messages migration:', error);
+      throw error;
+    }
+  }
+
+  private runUpgradeHistorySchemaMigration(): void {
+    try {
+      const migrationKey = 'migration_052_upgrade_history_schema';
+      const migrationCompleted = this.getSetting(migrationKey);
+
+      if (migrationCompleted === 'completed') {
+        logger.debug('✅ Upgrade history schema migration already completed');
+        return;
+      }
+
+      logger.debug('Running migration 052: Fix upgrade_history schema...');
+      upgradeHistorySchemaMigration.up(this.db);
+      this.setSetting(migrationKey, 'completed');
+      logger.debug('✅ Upgrade history schema migration completed successfully');
+    } catch (error) {
+      logger.error('❌ Failed to run upgrade history schema migration:', error);
       throw error;
     }
   }
@@ -5787,7 +5809,7 @@ class DatabaseService {
             ORDER BY timestamp DESC
             LIMIT 1
           )
-        `, [success, toNodeNum]);
+        `, [success ? 1 : 0, toNodeNum]);
       } else if (this.drizzleDbType === 'mysql' && this.mysqlPool) {
         await this.mysqlPool.query(`
           UPDATE auto_traceroute_log
@@ -8839,6 +8861,9 @@ class DatabaseService {
       // Run migration 051: Add decrypted_by column to messages
       await runMigration051Postgres(client);
 
+      // Run migration 052: Fix upgrade_history schema
+      await runMigration052Postgres(client);
+
       // Verify all expected tables exist
       const result = await client.query(`
         SELECT table_name FROM information_schema.tables
@@ -8902,6 +8927,9 @@ class DatabaseService {
 
       // Run migration 051: Add decrypted_by column to messages
       await runMigration051Mysql(pool);
+
+      // Run migration 052: Fix upgrade_history schema
+      await runMigration052Mysql(pool);
 
       // Verify all expected tables exist
       const [rows] = await connection.query(`

